sonar-scanner-engine/src/main/java/org/sonar/scanner/repository/user/UserRepositoryLoader.java

```diff
@@ -46,15 +50,32 @@ public UserRepositoryLoader(ScannerWsClient wsClient) {
     return parseUser(is);
   }
 
-  public Collection<ScannerInput.User> load(List<String> userLogins) {
+  public Collection<ScannerInput.User> load(Collection<String> userLogins) {
     if (userLogins.isEmpty()) {
       return Collections.emptyList();
     }
-    InputStream is = loadQuery(Joiner.on(',').join(Lists.transform(userLogins, new UserEncodingFunction())));
+    UserEncodingFunction userEncodingFunction = new UserEncodingFunction();
+    String encodedUserLogins = userLogins.stream()
+      .map(userEncodingFunction::apply)
+      .collect(Collectors.joining(","));
+    InputStream is = loadQuery(encodedUserLogins);
 
     return parseUsers(is);
   }
 
+  public Map<String, ScannerInput.User> map(Collection<String> userLogins) {
+    Set<String> userSet = new HashSet<>(userLogins);
+    Collection<ScannerInput.User> users = load(userLogins);
+    Map<String, ScannerInput.User> map = new HashMap<>();
+
+    for (ScannerInput.User user : users) {
+      if (userSet.contains(user.getLogin())) {
```

Comment:
Under what circumstances would `users` contain a user that was not in `userLogins` ?